test/e2e/test_image.py

```diff
@@ -95,3 +96,82 @@ def test_create_with_invalid_image_file(self):
 
         self.post_image(IMAGE_POST_METADATA_DATA_REQUIRED_VALUES_ONLY, "invalid_image.jpg")
         self.check_post_image_failed_with_detail(422, "File given is not a valid image")
+
+
+class ListDSL(CreateDSL):
+    """Base class for list tests."""
+
+    _get_response_image: Response
+
+    def get_images(self, filters: Optional[dict] = None) -> None:
+        """Gets a list of images with the given filters.
+
+        :param filters: Filters to use in the request."""
+        self._get_response_image = self.test_client.get("/images", params=filters)
+
+    def check_get_images_success(self, expected_images_get_data: list[dict]) -> None:
+        """
+        Checks that a prior call to `get_images` gave a successful response with the expected data returned.
+
+        :param expected_images_get_data: List of dictionaries containing the expected image data as would
+            be required for a `ImageSchema`.
+        """
+        assert self._get_response_image.status_code == 200
+        assert self._get_response_image.json() == expected_images_get_data
+
+
+class TestList(ListDSL):
+    """Tests for getting a list of images."""
+
+    def test_list_with_no_filters(self):
+        """
+        Test getting a list of all images with no filters provided.
+
+        Posts an image and expects it to be returned.
+        """
+
+        self.post_image(IMAGE_POST_METADATA_DATA_ALL_VALUES, "image.jpg")
+        expected_result = self._post_response_image.json()
```

Comment:
Could you add a `post_test_images` function that posts at least two images tor test the filtering on like we do for IMS e2e tests. Then you can return the actual post data and avoid having to use `self._post_response_image` directly.